Assets/MixedRealityToolkit.SDK/Features/Utilities/StabilizationPlaneModifier.cs

```diff
@@ -0,0 +1,390 @@
+            else
+            {
+                focusPointDistance = DefaultPlaneDistance;
+            }
+
+            float lerpPower = focusPointDistance > currentPlaneDistance ? LerpStabilizationPlanePowerFarther
+                                                                        : LerpStabilizationPlanePowerCloser;
+
+            // Smoothly move the focus point from previous hit position to new position.
+            currentPlaneDistance = Mathf.Lerp(currentPlaneDistance, focusPointDistance, lerpPower * deltaTime);
+
+            planePosition = gazeOrigin + (gazeDirection * currentPlaneDistance);
+
+#if UNITY_EDITOR
+            debugPlane.Center = planePosition;
+            debugPlane.Normal = -gazeDirection;
+#elif UNITY_WSA
+            HolographicSettings.SetFocusPointForFrame(planePosition, -gazeDirection, Vector3.zero);
+#endif
+        }
+
+        /// <summary>
+        /// Configures the stabilization plane to update based on a fixed distance away from you.
+        /// </summary>
+        private void ConfigureFixedDistancePlane(float deltaTime)
+        {
+            Vector3 gazeOrigin = GazeOrigin;
+            Vector3 gazeNormal = GazeNormal;
+
+            float lerpPower = DefaultPlaneDistance > currentPlaneDistance ? LerpStabilizationPlanePowerFarther
+                                                                          : LerpStabilizationPlanePowerCloser;
+
+            // Smoothly move the focus point from previous hit position to new position.
+            currentPlaneDistance = Mathf.Lerp(currentPlaneDistance, DefaultPlaneDistance, lerpPower * deltaTime);
+
+            planePosition = gazeOrigin + (gazeNormal * currentPlaneDistance);
+#if UNITY_EDITOR
+            debugPlane.Center = planePosition;
+            debugPlane.Normal = -gazeNormal;
+#elif UNITY_WSA
+            HolographicSettings.SetFocusPointForFrame(planePosition, -gazeNormal, Vector3.zero);
+#endif
+        }
+
+        /// <summary>
+        /// Tracks the velocity of the target object to be used as a hint for the plane stabilization.
+        /// </summary>
+        private Vector3 UpdateVelocity(float deltaTime)
+        {
+            // Roughly calculate the velocity based on previous position, current position, and frame time.
+            Vector3 velocity = (TargetOverride.position - targetOverridePreviousPosition) / deltaTime;
+            targetOverridePreviousPosition = TargetOverride.position;
+            return velocity;
+        }
+
+        /// <summary>
+        /// When in editor, draws a magenta quad that visually represents the stabilization plane.
+        /// </summary>
+        private void OnDrawGizmos()
+        {
+            if (Application.isPlaying && DrawGizmos && Mode != StabilizationPlaneMode.Off)
+            {
+                Gizmos.color = Color.green;
+                Gizmos.DrawWireMesh(debugMesh.GetComponent<MeshFilter>().sharedMesh, debugPlane.Center, Quaternion.LookRotation(debugPlane.Normal));
```

Comment:
Cache GetComponent<MeshFilter> instead of accessing every draw

Reply:
done